src/roiextractors/segmentationextractor.py

```diff
@@ -319,3 +323,120 @@ def write_segmentation(segmentation_extractor, save_path, overwrite=False):
             If True, the file is overwritten if existing (default False)
         """
         raise NotImplementedError
+
+
+class FrameSliceSegmentationExtractor(SegmentationExtractor):
+    """
+    Class to get a lazy frame slice.
+
+    Do not use this class directly but use `.frame_slice(...)`
+    """
+
+    extractor_name = "FrameSliceSegmentationExtractor"
+    is_writable = True
+
+    def __init__(
+        self,
+        parent_segmentation: SegmentationExtractor,
+        start_frame: Optional[int] = None,
+        end_frame: Optional[int] = None,
+    ):
+        self._parent_segmentation = parent_segmentation
+        self._start_frame = start_frame or 0
+        self._end_frame = end_frame or self._parent_segmentation.get_num_frames()
+        self._num_frames = self._end_frame - self._start_frame
+
+        self._image_masks = self._parent_segmentation._image_masks
+
+        parent_size = self._parent_segmentation.get_num_frames()
+        if start_frame is None:
+            start_frame = 0
+        else:
+            assert 0 <= start_frame < parent_size
+        if end_frame is None:
+            end_frame = parent_size
+        else:
+            assert 0 < end_frame <= parent_size
+        assert end_frame > start_frame, "'start_frame' must be smaller than 'end_frame'!"
+
+        super().__init__()
+        if getattr(self._parent_segmentation, "_times") is not None:
+            self._times = self._parent_segmentation._times[start_frame:end_frame]
+
+    def get_accepted_list(self) -> list:
```

Comment:
I guess this is for testing purposes only but what if the frames that you cut do not contain the rois here. I don't remember what spikeinterface does in this case. Most likely they don't check for this type of condition.

Reply:
How do you mean? With regards to motion correction?

Reply:
Currently there is no time mapping from frames to roi_ids, so that kind of thing is not currently possible or a 'problem'

Reply:
Right, that makes sense, I was thinking that in spikeinterface there are some units whose first spike is later in time and maybe it should be cut-out if you sub-frame but it seems that here that's not a problem as we have traces instead for each roi and image masks who don't really have an "time of first spike" concept.